### DISCUSSION TITLE
How to use query_facts to get dimensions={}

### DISCUSSION THREAD

**Discussion by jfraschilla** [dgunning/edgartools#118] [Q&A] [answered]
I am new to edgartools and have gotten some basic information from youtube.  I am using query_facts as shown
```
instance.query_facts(end_date=report_date, duration="3 months", concept="us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax", dimensions={})
```
which returns:
|     | concept                                                     |       value | units   |   decimals | start_date   | end_date   | period_type   | duration   | context_id   |   entity_id | dimensions                                                                       |
|----:|:------------------------------------------------------------|------------:|:--------|-----------:|:-------------|:-----------|:--------------|:-----------|:-------------|------------:|:---------------------------------------------------------------------------------|
|  62 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax | 61564000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-12         |  0000320193 | {'srt:ProductOrServiceAxis': 'us-gaap:ProductMember'}                            |
|  66 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax | 24213000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-16         |  0000320193 | {'srt:ProductOrServiceAxis': 'us-gaap:ServiceMember'}                            |
|  70 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax | 85777000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-20         |  0000320193 | {}                                                                               |
| 373 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax | 39296000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-59         |  0000320193 | {'srt:ProductOrServiceAxis': 'aapl:IPhoneMember'}                                |
| 377 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax |  7009000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-63         |  0000320193 | {'srt:ProductOrServiceAxis': 'aapl:MacMember'}                                   |
| 381 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax |  7162000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-67         |  0000320193 | {'srt:ProductOrServiceAxis': 'aapl:IPadMember'}                                  |
| 385 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax |  8097000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-71         |  0000320193 | {'srt:ProductOrServiceAxis': 'aapl:WearablesHomeandAccessoriesMember'}           |
| 389 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax | 24213000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-16         |  0000320193 | {'srt:ProductOrServiceAxis': 'us-gaap:ServiceMember'}                            |
| 393 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax | 85777000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-20         |  0000320193 | {}                                                                               |
| 716 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax | 37678000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-131        |  0000320193 | {'us-gaap:StatementBusinessSegmentsAxis': 'aapl:AmericasSegmentMember'}          |
| 724 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax | 21884000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-135        |  0000320193 | {'us-gaap:StatementBusinessSegmentsAxis': 'aapl:EuropeSegmentMember'}            |
| 732 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax | 14728000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-139        |  0000320193 | {'us-gaap:StatementBusinessSegmentsAxis': 'aapl:GreaterChinaSegmentMember'}      |
| 740 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax |  5097000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-143        |  0000320193 | {'us-gaap:StatementBusinessSegmentsAxis': 'aapl:JapanSegmentMember'}             |
| 748 | us-gaap:RevenueFromContractWithCustomerExcludingAssessedTax |  6390000000 | usd     |         -6 | 2024-03-31   | 2024-06-29 | duration      | 3 months   | c-147        |  0000320193 | {'us-gaap:StatementBusinessSegmentsAxis': 'aapl:RestOfAsiaPacificSegmentMember'} |

How can I query for the rows that have dimensions={}?
Why are there two identical rows?

Thanks

**Comment by dgunning** [comment, marked as answer]
This is actually a bug. Querying for empty dimensions should work with dimensions = {}.

I have done a fix and should be available soon (within a couple of days).

**Comment by jfraschilla** [comment]
@dgunning, Thanks for the quick reply and for your work on this very useful product.  What about the duplicate entries? Is it because it appears in multiple sections of the 10-Q?
